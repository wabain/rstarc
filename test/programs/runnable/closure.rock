MakeFnIndirect takes Value
    MakeFn takes NoArg
        Fn takes NewValue
            Put Value Into Ret
            Put NewValue Into Value
            Give back Ret
        (end function)
        Give back Fn
    (end function)
    Put MakeFn taking nothing Into FnLocal
    Give back FnLocal
(end function)

Put MakeFnIndirect taking "a" Into FnOne
Put MakeFnIndirect taking "b" Into FnTwo

Put FnOne taking 100 Into OutOne
Say OutOne

Put FnTwo taking FnOne Into OutTwo
Say OutTwo

Put FnTwo taking "x" Into OutTwo
Say OutTwo

Put FnOne taking 101 Into OutOne
Say OutOne
