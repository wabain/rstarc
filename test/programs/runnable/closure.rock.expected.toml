[tests]
enabled = ["ir", "run"]

[ir]
returncode = "1"
stderr = """
error: using non-local variable 'Value' in compiled code is not supported.

Try executing your program using 'rstarc run --interpret <program>' instead

4:              Put Value Into Ret
                    ^^^^^
"""

["run.exec"]
returncode = "1"
stderr = """
error: using non-local variable 'Value' in compiled code is not supported.

Try executing your program using 'rstarc run --interpret <program>' instead

4:              Put Value Into Ret
                    ^^^^^
"""

["run.interpret"]
returncode = "0"
stdout = """
a
b
[Function]
100
"""
